addons/l10n_ar_withholding/tests/test_withholding_ar_ri.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.exceptions import UserError
+from odoo.tests import tagged, Form
+from odoo import Command
+from odoo.tests.common import TransactionCase, Form
+
+
+@tagged('post_install', '-at_install')
+class TestL10nArWithholdingArRi(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref='ar_ri'):
+
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.tax_wth_seq_test_1 = cls.env['ir.sequence'].create({
+        'implementation': 'standard',
+        'name': 'tax wth test',
+        'padding': 8,
+        'number_increment' :1,            
+        })
+
+        cls.tax_wth_seq_test_2 = cls.env['ir.sequence'].create({
+        'implementation': 'standard',
+        'name': 'tax wth test',
+        'padding': 8,
+        'number_increment' :1,            
+        })
+
+        cls.tax_wth_test_1 = cls.env.ref('account.%i_ri_tax_withholding_iibb_caba_applied' % cls.env.company.id)
+
+        cls.tax_wth_test_1.write({
+            'l10n_ar_withholding_amount_type': 'untaxed_amount',
+            'amount': 1,                     
+            'amount_type': 'percent',
+        })
+        cls.tax_wth_test_2 = cls.env['account.tax'].create({
+            'description': 'wth_test_2',
+            'name': 'withholding test 2',
+            'l10n_ar_withholding': 'supplier',
+            'l10n_ar_withholding_amount_type': 'total_amount',
+            'l10n_ar_withholding_sequence_id': cls.tax_wth_seq_test_2.id,
+            'amount': 1,                     
+            'amount_type': 'percent',
+            'repartition_line_ids': [
+                Command.create({'document_type': 'invoice', 'repartition_type': 'base', 'tag_ids': []}),
+                Command.create({'document_type': 'invoice', 'repartition_type': 'tax', 'tag_ids': []}),
+                Command.create({'document_type': 'refund', 'repartition_type': 'base', 'tag_ids': []}),
+                Command.create({'document_type': 'refund', 'repartition_type': 'tax', 'tag_ids': []}),   
+            ]  
+        })
```

Comment:
why create all that in the test, when we might have it from the installation of the coa in the test company?

Reply:
done (refactor)